libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SmartStoreSDKManager.java

```diff
@@ -133,7 +133,8 @@ public static SmartStoreSDKManager getInstance() {
 
     @Override
     protected void cleanUp(UserAccount userAccount) {
-        if (userAccount != null && hasSmartStore(userAccount)) {
+        if (userAccount != null) {
```

Comment:
DBOpenHelper's openHelpers was not cleaned up on logout in other processes because of the hasSmartStore() check in SmartStoreSDKManager.cleanUp

As a result, after logout/login the sync process in SmartSyncExplorer would fail to write to sqlite with "error code 8: attempt to write a readonly database"